### PR TITLE
Instantiate CellAttachedData and CellAttachedDataSerializer

### PR DESCRIPTION
**Pull request by peterrum**
references https://github.com/dealii/dealii/issues/15698

introduced in #15576

FYI @pcafrica

### CODE REVIEW

File: source/grid/tria.inst.in [hunk 1]
```diff
@@ -19,5 +19,8 @@ for (deal_II_dimension : DIMENSIONS; deal_II_space_dimension : SPACE_DIMENSIONS)
   {
 #if deal_II_dimension <= deal_II_space_dimension
     template class Triangulation<deal_II_dimension, deal_II_space_dimension>;
+    template class CellAttachedData<deal_II_dimension, deal_II_space_dimension>;
+    template class CellAttachedDataSerializer<deal_II_dimension,
```
Comment (pcafrica) [inline source/grid/tria.inst.in:23]:
```suggestion
    template class internal::CellAttachedData<deal_II_dimension, deal_II_space_dimension>;
    template class internal::CellAttachedDataSerializer<deal_II_dimension,
```

**Comment by pcafrica** [comment]
Thank you @peterrum!

**Comment by kronbichler** [comment]
```
/Users/jenkins/jenkins/workspace/dealii-osx_PR-15700/build/source/grid/tria.inst:12:11: error: class template 'CellAttachedData' was previously declared as a struct template; this is valid, but may result in linker errors under the Microsoft C++ ABI [-Werror,-Wmismatched-tags]
 template class internal::CellAttachedData< 1 ,
          ^
/Users/jenkins/jenkins/workspace/dealii-osx_PR-15700/include/deal.II/grid/tria.h:339:10: note: previous use is here
  struct CellAttachedData
         ^
/Users/jenkins/jenkins/workspace/dealii-osx_PR-15700/build/source/grid/tria.inst:12:11: note: did you mean struct here?
 template class internal::CellAttachedData< 1 ,
          ^~~~~
          struct
```

**Review by kronbichler — APPROVED** [review]
Looks good to me.

**Review by kronbichler — APPROVED** [review]

**Pull request merged**